app/controllers/users_controller.rb

```diff
@@ -0,0 +1,30 @@
+class UsersController < ApplicationController
+  def show
+    @user = User.find_by(id: params[:id])
+    return if @user
+
+    flash[:danger] = t(".user_not_found")
+    redirect_to root_path
+  end
```

Comment:
```suggestion
  def show; end
```
Đưa logic bên trong ra thành method load_user và dùng trong before_action